aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -136,9 +162,36 @@ module aptos_framework::transaction_validation {
         );
         let balance = coin::balance<AptosCoin>(gas_payer);
         assert!(balance >= max_transaction_fee, error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT));
+
+        if (option::is_some(&required_deposit)) {
+            transaction_fee::burn_fee(gas_payer, option::extract(&mut required_deposit));
```

Comment:
we need a new error code if the max_transaction_fee != required_deposit, otherwise the abort will be `Unexpected prologue Move abort` https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/errors.rs#L64

Reply:
how does the comparison between `max_transaction_fee` and `required_deposit` matter?
did you mean a new error code for when there's not enough balance to deposit?

Reply:
new error code added